jobs/credhub/templates/application_spring.yml.erb

```diff
@@ -28,7 +28,7 @@
     properties['spring']['datasource'] = {
       'username' => username,
       'password' => password,
-      'url' => "jdbc:mariadb://#{host}:#{port}/#{database}?autoReconnect=true",
+      'url' => "jdbc:mariadb://#{host}:#{port}/#{database}?autoReconnect=true&socketTimeout=1200000",
```

Comment:
I would be more comfortable setting this timeout to one hour, since we don't know that we don't have slow queries that could take up to one hour.

Reply:
> At the very least, I'd like to see something like
> ```
> socketTimeoutInMilliseconds = 20 * 60 * 1000
> ```
`org.mariadb.jdbc.UrlParser.parseInternal` throws an Exception if the value is not an actual integer.

> would prefer to see the connection timeout set
Actually, I'd prefer not to have it to just set it to the same as the default value.  It gives me the impression that we are overriding the default when we are not.

> It took us weeks to understand what the default value was.
I don't know why.  It is well documented and easy to see in the code.

Reply:
> I would be more comfortable setting this timeout to one hour

Done

Reply:
>Actually, I'd prefer not to have it to just set it to the same as the default value. It gives me the impression that we are overriding the default when we are not.

I don't understand that argument. We *are* overriding the default value because we want it to be that specific value.

How do you envision anybody being confused in the future by that? How could someone make a mistake, and say "oh it was because I had the impression that the value was overriden!".

On the other hand, when you don't override the value, you give the impression that you don't care about what the value is. Which is not true.